From 854931b4d7326afb57e923b64bb8eca433956095 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Tue, 30 Apr 2013 13:50:23 +0800
Subject: MPEG4Extractor: Ignore invalid 'stts' chunk when playing H263 clips for MPEG4Extrator

BZ: 48862

Original patch:
32395 (BZ: 16520)

The sample clips with amr_wb audio track can't be played, because the audio track(amr_wb) contains
more than one `stts` chunks, and the first one is valid with version 0 and flag 0, others are invalid
with error version number and flag. MPEG4Extractor will iterate all the `stts` chunks, once met the
invalid `stts` chunk, the extractor will return error and exist. And return error is too much here, throw
out a warning is just enough.

Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: yes

Change-Id: I893ef8d95023b46e94724e4fa67b6818e0d298c5
Orig-Change-Id: I6d039fa72d904da0c8fc8d62ffec59f649de49ec
Signed-off-by: Weian Chen <weian.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/SampleTable.cpp |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/SampleTable.cpp b/media/libstagefright/SampleTable.cpp
index b560d84..75f63d5 100644
--- a/media/libstagefright/SampleTable.cpp
+++ b/media/libstagefright/SampleTable.cpp
@@ -330,7 +330,8 @@ status_t SampleTable::setTimeToSampleParams(
 
     if (U32_AT(header) != 0) {
         // Expected version = 0, flags = 0.
-        return ERROR_MALFORMED;
+        ALOGW("Expected version = 0, flags = 0");
+        return OK;
     }
 
     mTimeToSampleCount = U32_AT(&header[4]);
-- 
1.7.4.1

